src/loaders/png/tvgPngLoader.cpp

```diff
@@ -0,0 +1,52 @@
+#include <fstream>
+#include <string>
+#include "tvgLoaderMgr.h"
+#include "tvgPngLoader.h"
+// #include "png.h"
+
+PngLoader::~PngLoader()
+{
+    if (content) {
+        free((void*)content);
+        content = nullptr;
+    }
+}
+
+bool PngLoader::open(const string& path)
+{
+    png_image image;
+    image.version = PNG_IMAGE_VERSION;
+    image.opaque = NULL;
+
+    if (!png_image_begin_read_from_file(&image, path.c_str())) return false;
+    w = image.width;
+    h = image.height;
+    vw = w;
+    vh = h;
+    png_bytep buffer;
+    image.format = PNG_FORMAT_BGRA;
+    buffer = static_cast<png_bytep>(malloc(PNG_IMAGE_SIZE(image)));
+    if (!buffer) {
+        // out of memory, only time when libpng doesnt free its data
+        png_image_free(&image);
+        return false;
+    }
+    if (!png_image_finish_read(&image, NULL, buffer, 0, NULL)) return false;
+    content = reinterpret_cast<uint32_t*>(buffer);
```

Comment:
I think this part can be moved to read().

Reply:
I will move it if you want me to, although I think that it doesn't really matter in this case.

Reply:
Hi 
You can declare `png_image image` as a member field of PngLoader class.
PngLoader inherits Task (tvgTaskScheduler.h) and can execute `buffer = malloc(image)` and `png_..._read()` in thread. please see tvgSvgLoader class.